### PR TITLE
Move istio installation to terraform and fix the delete logic

### PR DESCRIPTION
**Pull request by raddaoui**

### CODE REVIEW

**Comment by raddaoui** [comment]
ok to test

**Comment by chrislovecnm** [comment]
Do we still have the ILB installed when it did not tear it down?

**Comment by raddaoui** [comment]
ok to test

**Comment by raddaoui** [comment]
ok to test

File: scripts/teardown.sh [hunk 1]
```diff
@@ -20,41 +20,33 @@ set -x
...
-              --filter "description:istio-system OR description:kube-system/dns-ilb") == "" ]]; do
+              --filter "description ~ istio-system.*ilb OR description:kube-system/dns-ilb") == "" ]]; do
   echo "Waiting for cluster to become ready for destruction..."
   sleep 10
 done
 
-until [[ $(gcloud --project="${ISTIO_PROJECT}" compute firewall-rules list \
-             --filter "name:k8s AND targetTags.list():gke-${ISTIO_CLUSTER}" \
-             --format "value(name)") == "" ]]; do
+until [[ $(gcloud --project="${ISTIO_PROJECT}" compute firewall-rules list --format yaml \
```
Comment (chrislovecnm) [inline scripts/teardown.sh]: Can we list this as a bug? This is just weird. Workaround for now

**Comment by chrislovecnm** [comment]
Test this please

**Comment by chrislovecnm** [comment]
Want to test this one more time

File: scripts/teardown.sh (hunk 1 above)
Comment (raddaoui) [reply scripts/teardown.sh]: this is not a workaround btw! these are the firewall rules that gets created by the mesh_expansion script and need to wait on all of them to get deleted. 
I am opening  a bug for another issue which I found and it's causing all this.

**Comment by raddaoui** [comment]
ok to test

**Comment by raddaoui** [comment]
Test this please

File: scripts/teardown.sh [hunk 2]
```diff
@@ -20,41 +20,40 @@ set -x
...
-    --project "$ISTIO_PROJECT" \
-    --zone "$ZONE"
+ISTIO_DIR="$ROOT/istio-${ISTIO_VERSION}"
 
 kubectl delete ns vm --ignore-not-found=true
 kubectl delete ns bookinfo --ignore-not-found=true
-kubectl delete svc dns-ilb -n kube-system --ignore-not-found=true
+
+# Uninstall the ILBs installed for mesh expansion
+kubectl delete -f "$ISTIO_DIR/install/kubernetes/mesh-expansion.yaml" --ignore-not-found=true
```
Comment (chrislovecnm) [inline scripts/teardown.sh]: We need to wait for this to complete.

Comment (raddaoui) [reply scripts/teardown.sh]: its already there! i only added a command to delete a couple of firewall rules that occasionally 
 are not deleted when you delete the cluster.

**Pull request merged**